crates/cargo-util-schemas/src/core/partial_version.rs

```diff
@@ -163,6 +159,22 @@ impl<'de> serde::Deserialize<'de> for PartialVersion {
     }
 }
 
+/// Error parsing a [`PartialVersion`].
+#[derive(Debug, thiserror::Error)]
+pub enum PartialVersionError {
+    #[error("unexpected version requirement, expected a version like \"1.32\"")]
+    VersionReq,
```

Comment:
Making the error type a `enum`, rather than a struct with a `Kind` element, limits our ability to expand this in the future (e.g. with backtraces which this changes loses)

Reply:
I've thought of an error struct wrapper, but realized that we might be happy to just publish breaking change.

Anyway, do you have a desired pattern of `ErrorKind` in mind?

Reply:
Made each error a tuple struct with a private error kind element.